subsys/bootloader/bl_storage/nrf_nv_counters.c

```diff
@@ -56,14 +56,15 @@ int32_t boot_nv_security_counter_update(uint32_t image_id, uint32_t img_security
 	 */
 	fih_int security_cnt;
 
-	fih_int fih_err = boot_nv_security_counter_get(image_id, &security_cnt);
+	fih_int fih_err;
 
-	if (fih_err != FIH_SUCCESS) {
-		return fih_err;
+	FIH_CALL(boot_nv_security_counter_get, fih_err, image_id, &security_cnt);
+	if (FIH_NOT_EQ(fih_err, FIH_SUCCESS)) {
+		return -1;
 	}
 
 	if (security_cnt == (uint16_t)img_security_cnt) {
-		return FIH_SUCCESS;
+		return 0;
```

Comment:
In the realm of mixing error codes from different domains.
From what I can tell this function returns BOOT_E<code> error codes, while the `set_monotonic_counter` function uses zephyr errno error codes.
If the codes carry meaning they can now be changed. 

So I think it would be more clean to also check and convert that error code as well.

Reply:
How can you tell that? Could you explain please? I only see that it says "0 on success; nonzero on failure." in the signagure.

Reply:
In the bootloader code we usually return BOOT_E<code> error code, while the function that you're calling here returns E<code> error codes.
One returns 'int' the other 'uint32_t', so there is also that.
So the value that they return are in different domains, meaning that the code cannot be correctly interpreted, other than 0 being success.